numpy/core/include/numpy/npy_math.h

```diff
@@ -210,8 +210,10 @@ double npy_spacing(double x);
         (sizeof (x) == sizeof (long double) ? _npy_signbit_ld (x) \
          : sizeof (x) == sizeof (double) ? _npy_signbit_d (x) \
          : _npy_signbit_f (x))
+    #define true_npy_signbit(x) (signbit((x)) != 0 ? 1 : 0)
 #else
     #define npy_signbit(x) signbit((x))
+    #define true_npy_signbit(x) (signbit((x)) != 0 ? 1 : 0)
```

Comment:
these macros in headers are problematic in general, as they are usually only macros inside numpy and slow functions outside as we don't know the used compilers
maybe its better to just fix the internal user and not add more macros